### PR TITLE
Don't duplicate records in esf menu when filteringIgnoreCase is true.

### LINKED ISSUE #8162: excelStyleFiltering search duplicates records when filteringIgnoreCase is set to true
## Description  
If we have the following data:
[{ProductID: 0, ProductName: "Coffee", ...}, {ProductID:1, ProductName: "CoffeE", ...}, ...]
and we have set filteringIgnoreCase to true for the igxColumn with field: ProductName,
within the excelStyleFiltering search list we observe both values: Coffee and CoffeE

 * igniteui-angular version: 10.1.x, 9.1.x
 * browser: any

## Steps to reproduce  

1. Open the following sample: https://stackblitz.com/edit/excel-style-filtering-ignore-case
2. Open the excelStyleFiltering for the ProductName column

## Result  
Both records exist: Coffee and CoffeE

## Expected result  
To have a single entry for Coffee and CoffeE

### PR DESCRIPTION
**Pull request by onlyexeption**
Closes #8162

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Review by astaev — DISMISSED** [review]

File: projects/igniteui-angular/src/lib/grids/filtering/excel-style/grid.excel-style-filtering.component.ts [hunk 1]
```diff
@@ -599,6 +599,13 @@ export class IgxGridExcelStyleFilteringComponent implements OnDestroy {
 
         this.uniqueValues.forEach(element => {
             if (element !== undefined && element !== null && element !== '') {
+                if (this.column.dataType === DataType.String &&
+                    this.column.filteringIgnoreCase &&
+                    this.listData.length > 0 &&
+                    this.listData.map(el => el.label.toLowerCase()).indexOf(element.toLowerCase()) !== -1) {
+                        return;
+                    }
+
```
Comment (DiyanDimitrov) [inline projects/igniteui-angular/src/lib/grids/filtering/excel-style/grid.excel-style-filtering.component.ts:608]: This doesn't seems to be the best place to apply the `filteringIgnoreCase`, it is better to apply it in the `generateUniqueValues` method. Also this solution is not the best in terms of performance. You can see a performance comparison [here](https://stackoverflow.com/a/43046408) and an implementation for case insensitive scenarios [here](https://stackoverflow.com/a/48731445).

**Review by DiyanDimitrov — CHANGES_REQUESTED** [review]

File: projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts [hunk 2]
```diff
@@ -4384,10 +4384,36 @@ describe('IgxGrid - Filtering actions - Excel style filtering #grid', () => {
...
+            const column = grid.getColumnByName('AnotherField');
+            expect(column.filteringIgnoreCase).toBeTrue();
+
+            GridFunctions.clickExcelFilterIconFromCode(fix, grid, 'AnotherField');
+            tick(100);
+            fix.detectChanges();
+
+            const searchComponent = GridFunctions.getExcelStyleSearchComponent(fix);
+            const listItems = GridFunctions.getExcelStyleSearchComponentListItems(fix, searchComponent);
+            expect(listItems.length).toBe(3, 'incorrect rendered list items count');
```
Comment (DiyanDimitrov) [inline projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts:4397]: It is good to check the item labels too to be sure the first occurrence is used.

File: projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts [hunk 3]
```diff
@@ -4384,10 +4384,36 @@ describe('IgxGrid - Filtering actions - Excel style filtering #grid', () => {
...
+            column.filteringIgnoreCase = false;
+            expect(column.filteringIgnoreCase).toBeFalse();
 
+            GridFunctions.clickExcelFilterIconFromCode(fix, grid, 'AnotherField');
+            tick(100);
+            fix.detectChanges();
+
+            const searchComponent = GridFunctions.getExcelStyleSearchComponent(fix);
+            const listItems = GridFunctions.getExcelStyleSearchComponentListItems(fix, searchComponent);
+            expect(listItems.length).toBe(5, 'incorrect rendered list items count');
```
Comment (DiyanDimitrov) [inline projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts:4411]: Check the item labels too.

File: projects/igniteui-angular/src/lib/grids/filtering/excel-style/grid.excel-style-filtering.component.ts [hunk 4]
```diff
@@ -501,7 +501,15 @@ export class IgxGridExcelStyleFilteringComponent implements OnDestroy {
     }
 
     private generateUniqueValues(columnValues: any[]) {
-        this.uniqueValues = Array.from(new Set(columnValues));
+        if (this.column.dataType === DataType.String && this.column.filteringIgnoreCase) {
+            const filteredUniqueValues = columnValues.map(s =>
+                s.toLowerCase()).reduce((map, val, i) => map.get(val) ? map : map.set(val, columnValues[i]),
+                new Map);
```
Comment (DiyanDimitrov) [inline projects/igniteui-angular/src/lib/grids/filtering/excel-style/grid.excel-style-filtering.component.ts:507]: Use optional chaining for null values.
```suggestion
            const filteredUniqueValues = columnValues.map(s => s?.toLowerCase()).
                reduce((map, val, i) => map.get(val) ? map : map.set(val, columnValues[i]),
                new Map);
```

**Review by DiyanDimitrov — CHANGES_REQUESTED** [review]

**Review by DiyanDimitrov — APPROVED** [review]

**Pull request merged**